### PR TITLE
Fix issue with logging each cookie as separate context

### LINKED ISSUE #31471: [Issue] Fix issue with logging each cookie as separate context
This issue is automatically created based on existing pull request: magento/magento2#31334: Fix issue with logging each cookie as separate context

### Description (*)
When the number of cookies becomes more than 50 the warning log is written: "Unable to send the cookie. Maximum number of cookies would be exceeded." And each cookie is converting to a separate log context. And in the case of using the Graylog app, it is converting to a separate field.
By this pull request, the $_COOKIE array is converting to a string and writing as 'cookie' context while writing the log

### Manual testing scenarios (*)

1. Exceed the number of cookies > 50
1. Watch the log

### Preconditions 
Magento 2.4-develop

### Steps to reproduce 
1. Open web browser Firefox
1. Navigate to `about:config`
1. Find `dom.storage.enabled` option and change to *false*
1. Open Storefront in a new tab
1. Duplicate the tab, open the same page in another tab
1. Duplicate the tabs several times (3-4 new tabs)
1. Open `debug.log`  file from *./var/log* folder 
1. Find the text *"main.WARNING: Unable to send the cookie. Maximum number of cookies would be exceeded"*

### Actual Result
unable*to send*coockie

### Expected Result
unable*to send_coockie*after

### PR DESCRIPTION
**Pull request by aleksinoleg**
### Description (*)
When the number of cookies becomes more than 50 the warning log is written: "Unable to send the cookie. Maximum number of cookies would be exceeded." And each cookie is converting to a separate log context. And in the case of using the Graylog app, it is converting to a separate field.
By this pull request, the $_COOKIE array is converting to a string and writing as 'cookie' context while writing the log

### Manual testing scenarios (*)

1. Exceed the number of cookies > 50
2. Watch the log

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#31471: Fix issue with logging each cookie as separate context

### CODE REVIEW

**Comment by aleksinoleg** [comment]
This PR is https://github.com/magento/magento2/pull/31259 variant for 2.4

**Comment by ihor-sviziev** [comment]
@magento run all tests

File: lib/internal/Magento/Framework/Stdlib/Cookie/PhpCookieManager.php [hunk 1]
```diff
@@ -212,7 +212,21 @@ private function checkAbilityToSendCookie($name, $value)
...
+                            array_map(
+                                function ($v, $k) {
+                                    return sprintf("%s='%s'", $k, $v);
+                                },
+                                $_COOKIE,
+                                array_keys($_COOKIE)
+                            )
+                        )
+                    ]
+                )
```
Comment (ihor-sviziev) [inline lib/internal/Magento/Framework/Stdlib/Cookie/PhpCookieManager.php]: We don't really need to have array_merge here. Could you remove it?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
Finally, someone will fix that! 👍

We also were having this issue - we're sending logs to some Elasticsearch based tool, and it was hunting because of too many fields (each key is sent as a separate key). Also, it doesn't have any value for sending all cookies as separate values, so PR, in general, looks good to me.

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8557](https://jira.corp.magento.com/browse/ENGCOM-8557) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

### Manual testing scenario 
1. Open web browser Firefox
2. Navigate to `about:config`
3. Find `dom.storage.enabled` option and change to _false_
4. Open Storefront in a new tab
5. Duplicate the tab, open the same page in another tab
6. Duplicate the tabs several times (3-4 new tabs)
7. Open `debug.log`  file from _./var/log_ folder 
8. Find the text _"main.WARNING: Unable to send the cookie. Maximum number of cookies would be exceeded"_

**BEFORE** switching to the PR
unable_to send_coockie

**AFTER** applying changes from the PR,
unable_to send_coockie_after

**Comment by engcom-Bravo** [comment]
@magento create an issue

**Comment by engcom-Bravo** [comment]
@magento create issue

**Pull request merged**